src/AMAMM.sol

```diff
@@ -110,6 +111,14 @@ contract AMAMM is IAmAmm {
         _updateLastUpdatedEpoch(id, _epoch);
     }
 
+    function depositToken(PoolId id, address depositor, uint40 _epoch) internal returns (uint256) {
+        uint256 amount = _getDeposit(id, _epoch);
```

Comment:
I think the amount should be more flexible as a parameter rather than fixed from `_getDeposit` as the user could have existing user balances that only require a little bit of deposit.